src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -1155,6 +1155,14 @@ private ProjectInstance LoadProjectFromFile()
             // Get the hosted ISdkResolverService.  This returns either the MainNodeSdkResolverService or the OutOfProcNodeSdkResolverService depending on who created the current RequestBuilder
             ISdkResolverService sdkResolverService = _componentHost.GetComponent(BuildComponentType.SdkResolverService) as ISdkResolverService;
 
+            // Use different project load settings if the build request indicates to do so
+            ProjectLoadSettings projectLoadSettings = _componentHost.BuildParameters.ProjectLoadSettings;
+
+            if (_requestEntry.Request.BuildRequestDataFlags.HasFlag(BuildRequestDataFlags.IgnoreMissingEmptyAndInvalidImports))
```

Comment:
Since there is already a BuildParameters.ProjectLoadSettings, why bother with an additional BuildParameters.IgnoreMissingEmptyAndInvalidImports bool which then has to be translated back to ProjectLoadSettings? Instead, in Xmake.cs, you could compose the ProjectLoadSettings directly.

Reply:
Because you only get one BuildParameters between `BeginBuild` and `EndBuild` and we're queuing two build requests (Restore, Build).

Reply:
Mkay, so this is to allow finer granularity within a single session.
Regarding restore and build, I don't remember, why aren't they running in separate begin / end build sessions? Seems like we'd want to keep them as separate as possible.